src/main/java/com/exadel/discountwebapp/location/service/LocationService.java

```diff
@@ -49,14 +54,20 @@ public Location findEntityById(Long id) {
     @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
     public List<CityResponseVO> findAllCitiesByCountryCode(String countryCode) {
         List<Location> locations = locationRepository.findAllByCountry_CountryCode(countryCode);
```

Comment:
We are retrieving too much data here (which is not needed). In the nearest future we can refactor it to retrieve only what we need (custom projections?)

Reply:
Added protection, but I'm not sure if it's done right.